python/dgl/graphbolt/sampled_sub_graph.py

```diff
@@ -0,0 +1,172 @@
+"""Heterogeneous sampled subgraph."""
+# pylint: disable= invalid-name
+from collections.abc import Mapping
+from typing import Dict, Tuple, Union
+
+import torch
+
+
+class SampledSubGraph:
```

Comment:
Since this class is purely for storing data, can we use `@dataclass` or `NamedTuple`?

Reply:
Also change `SubGraph` to `Subgraph`.

Reply:
fixed